tests/installation/disk_activation.pm

```diff
@@ -26,79 +33,91 @@ sub run {
     if (check_var('S390_DISK', 'ZFCP')) {           # use zfcp as install disk
         assert_screen 'disk-activation-zfcp';
         send_key 'alt-z';                           # configure ZFCP disk
-        assert_screen 'zfcp-disk-management';
-        send_key 'alt-a';
-        assert_screen 'zfcp-add-device';
-        send_key $cmd{next};
+                                                    # workaround for bsc#1055871
+        if (!check_screen('zfcp-disk-management', 5)) {
+            workaround_broken_disk_activation;
+        }
+        else {
+            assert_screen 'zfcp-disk-management';
+            send_key 'alt-a';
+            assert_screen 'zfcp-add-device';
+            send_key $cmd{next};
 
-        # use allow_lun_scan
-        assert_screen 'zfcp-popup-scan';
-        send_key 'alt-o';
+            # use allow_lun_scan
+            assert_screen 'zfcp-popup-scan';
+            send_key 'alt-o';
 
-        assert_screen 'zfcp-disk-management';
-        assert_screen 'zfcp-activated';
-        send_key $cmd{next};
-        wait_still_screen 5;
+            assert_screen 'zfcp-disk-management';
+            assert_screen 'zfcp-activated';
+            send_key $cmd{next};
+            wait_still_screen 5;
+        }
     }
     else {    # use default DASD as install disk
         assert_screen 'disk-activation', 15;
         send_key 'alt-d';    # configure DASD disk
-        assert_screen 'dasd-disk-management';
-
-        # we need to type backspace to delete the content of the input field in textmode
-        if (check_var("VIDEOMODE", "text")) {
-            send_key 'alt-m';    # minimum channel ID
-            for (1 .. 9) { send_key "backspace"; }
-            type_string '0.0.0150';
-            send_key 'alt-x';    # maximum channel ID
-            for (1 .. 9) { send_key "backspace"; }
-            type_string '0.0.0150';
-            send_key 'alt-f';    # filter button
-            assert_screen 'dasd-unselected';
-            send_key 'alt-s';    # select all
-            assert_screen 'dasd-selected';
-            send_key 'alt-a';    # perform action button
-            assert_screen 'action-list';
-            send_key 'alt-a';    # activate
+                             # workaround for bsc#1055871
+        if (!check_screen('dasd-disk-management', 5)) {
+            workaround_broken_disk_activation;
```

Comment:
these lines are the actual change

Reply:
Let's then add this for sle 15, so if needle needs to be updated we don't use workaround. Otherwise LGTM.